tests/test_wcsfit.py

```diff
@@ -0,0 +1,547 @@
+# This file is part of drp_tasks
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import os.path
+import numpy as np
+import yaml
+import astropy.units as u
+import pandas as pd
+import wcsfit
+
+import lsst.utils
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwgeom
+import lsst.drp.tasks.wcsfit as lsst_wcsfit
+from lsst.daf.base import PropertyList
+from lsst.daf.butler import DimensionUniverse, DatasetType, DatasetRef, StorageClass
+from lsst.meas.algorithms import ReferenceObjectLoader
+from lsst import sphgeom
+import lsst.geom
+
+
+class MockInputVisitSummary():
+    """Object that provides a minimal inputVisitSummary table
+    """
+    def __init__(self, visit):
+
+        # Set up the schema for the output catalogs
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='L', doc='Visit number')
+
+        catalog = afwTable.ExposureCatalog(schema)
+        catalog.resize(len(self.detectors))
+        catalog['visit'] = visit
+
+
+class MockInputCatalogRef():
```

Comment:
There is a better way of doing this, with an `lsst.pipe.base.InMemoryDatasetHandle`.

Reply:
`lsst.pipe.base.InMemoryDatasetHandle` doesn't work because I need a `.ref` property. I played around with different methods of making something that would work in `lsst.meas.algorithms.ReferenceObjectLoader`, but I gave up and did this. I guess adding `.ref` to `InMemoryDatasetHandle` would be another option.

Reply:
Upon some follow-up, this should _not_ have a public `ref` property at all.  I'm curious what was throwing an error, or is it something just in the testing here?

Reply:
But I realize this is a bit outside of this ticket.  However, I think that we have some not great test code that has just been cargo culted around.  It might need a follow-up ticket.

Reply:
If a `PipelineTask.run()` method somewhere needs to know the `DatasetRef` of the dataset then that seems very wrong. I can't think of a reason a `run()` method would ever need to know that. It's public because it's a dataclass and so needs to be passed in to the constructor. Do we know what code is trying to look at the ref and why it's wanting to know?

Reply:
Why do you have all this infrastructure for making fake catalogs and datasetRefs, instead of using the testdata_jointcal package that has butler datasets ready to go?

@timj : not a topic for discussion on this ticket, but `ReferenceObjectLoader` needs the refs in its `__init__`.

Reply:
@parejkoj I think that the idea is with sim data here there are additional tests that are possible when you know the truth rather than for the real data in testdata_jointcal.

Also, the `ReferenceObjectLoader` needs the deferred dataset handles, not the `ref`.  I just did a quick modification locally to `test_wcsfit.py` and (a) added `from lsst.pipe.base import InMemoryDatasetHandle`; (b) deleted `MockInputCatalogRef`; (c) changed `inputCatalogRef = MockInputCatalogRef(visitSourceTable, dataId={"visit": visit})` to `inputCatalogRef = InMemoryDatasetHandle(visitSourceTable, storageClass="DataFrame", dataId={"visit": visit})`; (d) changed `deferredRefCat = MockInputCatalogRef(refCat, ref=refDataId.ref)` to `deferredRefCat = InMemoryDatasetHandle(refCat, storageClass="SourceCatalog")` and the tests ran fine.

Reply:
@parejkoj and @erykoff, I had to remind myself why I'm not using `testdata_jointcal`. `testdata_jointcal` would work in place of the included test data. Realistic WCSs and observation parameters are all that are needed; the source tables are simulated. However, I was specifically asked not to use `testdata_jointcal` because it has `obs_subaru` and other obs packages as dependencies, and we didn't want to tie `drp_tasks` to all of those packages.

Reply:
You would make it an optional dependency of this package, not a required one, and then have tests that check for its existence (as is done in fgcmcal and jointcal).

If you're going to not use testdata_jointcal (but I really think you should, since we've characterized that data pretty well), you need to move the test files out of this package and into an optional test data package anyway; we can't have large FITS files in code packages, as you're doing here.

Reply:
The `wcsfit.Instrument` is not related to the `obs.base.Instrument`. It is just a `gbdes` class that holds basic information, like the number of detectors and their dimensions (in pixels). The detector layout comes from the input WCS.

Reply:
Where does the number of detectors and their dimensions come from then? You should be getting that from the  Camera object.

Reply:
Re-asking my question above, as it probably got lost in the noise: Where does the number of detectors and their dimensions (that go into the wcsfit.Instrument object) come from? You should be getting that from the Camera object.

Reply:
The number and dimensions of the detectors comes from the `visitSummary` tables. The layout comes from the input wcs, which is also in the `visitSummary` tables.